### DISCUSSION TITLE
Does Quarto support themeVariables for Mermaid diagrams?

### DISCUSSION THREAD

**Discussion by FeralFlora** [quarto-dev/quarto-cli#5938] [Q&A] [answered]
### Description

I've been trying to change the text color of nodes in a LR Mermaid flowchart using `themeVariables` as documented by Mermaid.js [here](https://mermaid.js.org/config/theming.html#customizing-themes-with-themevariables). My goal is to make the text lighter, because the auto-themed colors based on the Moon Reveal.js theme are not very readable.

However, I cannot get any changes to occur at all, so it seems Quarto is not recognizing `themeVariables`, or am I making some mistake? 

Here's the code I was trying:

```
%%{init: {
  "theme": "base",
  "themeVariables": {
    "primaryTextColor": "#FFFFFF"
  }
}}%%
```

Do I need to use CSS instead?

**Comment by cscheid** [comment]
Have you tried using our [mermaid theming options](https://quarto.org/docs/authoring/diagrams.html#mermaid-theming)?

**Reply by FeralFlora** [reply]
Yes, I could achieve the color change with CSS. My previous attempts were inspired by this comment: https://github.com/quarto-dev/quarto-cli/issues/2165#issuecomment-1229193146 and the Mermaid.js documentation.

That example can change the theme as you would with the YAML option, but the `base` theme that works with `themeVariables` doesn't work.

Btw, I think the Moon theme would be more readable by default if the primaryTextColor matched the H2 heading color or the body text color. There is little contrast with the standard blue on blue.

Inherited:
![image](https://github.com/quarto-dev/quarto-cli/assets/33608083/a3136976-f9a6-4050-be40-238d14c059f6)
With CSS:
![image](https://github.com/quarto-dev/quarto-cli/assets/33608083/98396573-3279-41e7-89dc-a52f7e3c47b3)

**Reply by cscheid** [reply, marked as answer]
> That example can change the theme as you would with the YAML option, but the base theme that works with themeVariables doesn't work.

A general problem with mermaid is that it's impossible for us to determine exactly what variables we need to support on what charts, because they don't have a consistent theming system. I believe your report, but we cannot simultaneously support our theming system (which does at least make an attempt at consistency) and theirs, and we think ours is better. Unfortunately, that is not a bug we can fix.

**Reply by FeralFlora** [reply]
Alright, thanks for the clarification. I was mostly just curious about how misguided my failed attempt was - especially since that init approach in the linked comment actually works for built-in themes. I see that I can achieve my customization objectives with CSS as necessary, as you describe in the mermaid theming docs.
